pallets/maintenance-mode/src/lib.rs

```diff
@@ -81,6 +89,34 @@ pub mod pallet {
 		/// able to return to normal mode. For example, if your MaintenanceOrigin is a council, make
 		/// sure that your councilors can still cast votes.
 		type MaintenanceOrigin: EnsureOrigin<Self::Origin>;
+		/// The DMP handler to be used in normal operating mode
+		type NormalDmpHandler: DmpMessageHandler;
+		/// The DMP handler to be used in maintenance mode
+		type MaintenanceDmpHandler: DmpMessageHandler;
+		/// The XCMP handler to be used in normal operating mode
+		type NormalXcmpHandler: XcmpMessageHandler;
+		/// The XCMP handler to be used in maintenance mode
+		type MaintenanceXcmpHandler: XcmpMessageHandler;
+		/// The OnIdle hooks that should be called on normal operating mode
+		type NormalOnIdle: OnIdle<Self::BlockNumber>;
```

Comment:
```suggestion
		/// The OnIdle hooks that should be called on normal operating mode
		/// In most cases this should be `AllPallets`
		type NormalOnIdle: OnIdle<Self::BlockNumber>;
```

Most reuntime developers don't have a good underrstanding of what `AllPallets` is or how it should be used. They just manage to get it wired into the executive because they copy from the template. It might be worth making that known here.

It might also be a reasonable choice to remove these `NormalOnWhatever` types, and replace them with a  single
```rust
/// The `AllPallets` type generated by `construct_runtime!`. This is necessary for running frame hooks like `OnInitialize` and others
/// when the chain is not in maintenance mode.
type AllPallets: OnIdle<Self::BlockNumber> + OnInitialize<Self::BlockNumber + ...;
```

My suggestion is definitely less flexible, but also harder to configure incorrectly. I'm fine either way.

Reply:
Oh, but if we do keep the multiple `NormalOnWhatever` types, it would be nice to put them in block execution order, which is:
* runtime upgrade
* initialize
* post_inherent (doesn't exist yet, but shawn is working on it)
* idle
* finalize

Reply:
I initially had it like this, but this implies we need to create a custom struct in the runtime that matches our Maintenance executives (i.e., allPallets in all hooks except for OnIdle, which should be the DoNothingOnIdle). I felt this made the code in our runtime less clean, and that is the reason why I switched to this option. I am fine with either though

Reply:
I agree that it would be better to simplify the config. Remove the associated types for normal mode and replace them with an associated AllPallets type.
It doesn't make the pallet less flexible actually, because it is still possible to submit any struct that implements the hooks.

Reply:
Done

Reply:
Um, it seems to me that the proposal was rather to name the associated type `AllPallets` (instead of `NormalExecutiveHooks`), with an explanation in ructdoc of what AllPallets is, in order to simplify the possible future use of this pallet by other projects.

Reply:
So the suggestion is to change the name NormalExecutiveHooks for AllPallets? I find that name not descriptive of what the assocaited type should do, specially in the (not likely) situation where users want to configure this for normal operation. A clear example: what if, for whatever reason, I decice to disable onIdle hooks for xcm in normal operation?

Reply:
So if I understand correctly, you want to maintenanceOnIdle, maintenanceOnFinalize, etc associated types but for normal have a singe AllPallets type?

Reply:
Yes I understand, in this case can you at least indicate in the rustdoc associated with this associated type that we have to use `AllPallets` if we don't want to modify the hooks behavior?

Reply:
Absolutely, yes I can do that

Reply:
> So if I understand correctly, you want to maintenanceOnIdle, maintenanceOnFinalize, etc associated types but for normal have a singe AllPallets type?

I don't, but @JoshOrndorff I don't know.

Reply:
Ok so now we have single MaintenanceExecutiveHooks and NormalExecutiveHooks, which need to implement all the hooks. I can easily revert any of the changes, so let me know if the code feels more/less clear now :)